presto-main/src/main/java/com/facebook/presto/operator/aggregation/fixedhistogram/FixedHistogramUtils.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.fixedhistogram;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+class FixedHistogramUtils
+{
+    private FixedHistogramUtils() {}
+
+    public static void verifyParameters(int bucketCount, double min, double max)
+    {
+        checkArgument(bucketCount >= 2, "bucketCount %s must be at least 2", bucketCount);
+        checkArgument(min < max, "min %s must be smaller than max %s ", min, max);
+    }
+
+    public static int getIndexForValue(int bucketCount, double min, double max, double value)
+    {
+        checkArgument(
+                value >= min && value <= max,
+                "value must be within range [%s, %s]", min, max);
```

Comment:
What's the reasoning behind an inclusive range on both ends? It makes the handling a little more complicated and not very conventional from my (very limited) experience. Would love to learn about it.

Reply:
That's an excellent point. The [inclusive, exclusive) range makes a lot of sense in discrete ranges. For floating points, it could be awkward. Say that someone wants to bucket a probability (which is extremely common - all of classification metrics do that, for example), then how do you use it there? [0, 1.1)? [0, 1.0000000000000000001)? In practice, people will set 0 and 1 for the bounds, because that's how a probability is defined, and it'll be interpreted as [0, 1), which is wrong. The problem is that "next number" is not that simple for floating points.

Reply:
But if so happened that you get a value equal to the upper bound of a bucket, it would only belong to one bucket right. Even for double it can still happen that a value is exactly the same as the upper bound. So for all values that's equal to the upper bound of a bucket, it will be in this bucket or the next? Why would that behavior be different on one end from on the other?

Reply:
Yeah, that's a good point. I don't mind changing it to [begin, end).

Reply:
@rongrong Thanks for your comment! Updated it to be left-inclusive right-exclusive.